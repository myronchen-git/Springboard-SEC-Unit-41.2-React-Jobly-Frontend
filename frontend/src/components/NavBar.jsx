import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';

// ==================================================

function NavBar({ isLoggedIn = false }) {
  return (
    <nav className="NavBar">
      <NavLink to="/">Home</NavLink>
      {isLoggedIn ? (
        <>
          <NavLink to="/companies" end>
            Companies
          </NavLink>
          <NavLink to="/jobs" end>
            Jobs
          </NavLink>
          <NavLink to="/profile">Profile</NavLink>
        </>
      ) : (
        <>
          <NavLink to="/login">Login</NavLink>
          <NavLink to="/signup">Sign Up</NavLink>
        </>
      )}
    </nav>
  );
}

// ==================================================

NavBar.propTypes = {
  isLoggedIn: PropTypes.bool.isRequired,
};

export default NavBar;
