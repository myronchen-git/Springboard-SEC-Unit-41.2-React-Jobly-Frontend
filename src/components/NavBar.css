.NavBar {
  flex: 0 1 auto;
}
