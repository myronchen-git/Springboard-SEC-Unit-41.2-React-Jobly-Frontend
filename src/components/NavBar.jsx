import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

// ==================================================

/**
 * Displays the navigation bar.
 *
 * @param {Object} props - React component properties.
 * @param {String} props.username - Username of the logged in user, otherwise
 *   falsy.
 * @param {Function} props.logout - Logs out the current user.
 */
function NavBar({ username, logout }) {
  return (
    <nav className="NavBar">
      <NavLink to="/">Home</NavLink>
      {username ? (
        <>
          <NavLink to="/companies" end>
            Companies
          </NavLink>
          <NavLink to="/jobs" end>
            Jobs
          </NavLink>
          <NavLink to="/profile">Profile</NavLink>
          <NavLink to="/" onClick={logout}>
            Logout {username}
          </NavLink>
        </>
      ) : (
        <>
          <NavLink to="/login">Login</NavLink>
          <NavLink to="/signup">Sign Up</NavLink>
        </>
      )}
    </nav>
  );
}

// ==================================================

NavBar.propTypes = {
  username: PropTypes.string,
  logout: PropTypes.func.isRequired,
};

export default NavBar;
