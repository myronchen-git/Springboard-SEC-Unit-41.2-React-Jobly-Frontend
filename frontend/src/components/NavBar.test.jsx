import { fireEvent, render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { beforeEach, describe, expect, it, vi } from 'vitest';

import NavBar from './NavBar';

// ==================================================

describe('NavBar', () => {
  const username = 'testuser';
  const mockLogout = vi.fn();

  beforeEach(() => {
    mockLogout.mockReset();
  });

  it('renders', () => {
    render(
      <MemoryRouter>
        <NavBar username={username} logout={mockLogout} />
      </MemoryRouter>
    );
  });

  it('matches snapshot.', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <NavBar username={username} logout={mockLogout} />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('displays correct links when logged in.', () => {
    // Act
    const { queryByText, getByText } = render(
      <MemoryRouter>
        <NavBar username={username} logout={mockLogout} />
      </MemoryRouter>
    );

    // Assert
    expect(getByText('Home')).toBeVisible();
    expect(getByText('Companies')).toBeVisible();
    expect(getByText('Jobs')).toBeVisible();
    expect(getByText('Profile')).toBeVisible();
    expect(getByText(`Logout ${username}`)).toBeVisible();
    expect(queryByText('Login')).not.toBeInTheDocument();
    expect(queryByText('Sign Up')).not.toBeInTheDocument();

    expect(mockLogout).not.toHaveBeenCalled();
  });

  it('displays correct links when not logged in.', () => {
    // Arrange
    const username = undefined;

    // Act
    const { queryByText, getByText } = render(
      <MemoryRouter>
        <NavBar username={username} logout={mockLogout} />
      </MemoryRouter>
    );

    // Assert
    expect(getByText('Home')).toBeVisible();
    expect(queryByText('Companies')).not.toBeInTheDocument();
    expect(queryByText('Jobs')).not.toBeInTheDocument();
    expect(queryByText('Profile')).not.toBeInTheDocument();
    expect(queryByText(`Logout`, { exact: false })).not.toBeInTheDocument();
    expect(getByText('Login')).toBeVisible();
    expect(getByText('Sign Up')).toBeVisible();

    expect(mockLogout).not.toHaveBeenCalled();
  });

  it('logs out a user.', () => {
    // Arrange
    const { getByText } = render(
      <MemoryRouter>
        <NavBar username={username} logout={mockLogout} />
      </MemoryRouter>
    );

    // Act
    const logoutBtn = getByText('Logout', { exact: false });
    fireEvent.click(logoutBtn);

    // Assert
    expect(mockLogout).toHaveBeenCalledOnce();
  });
});
