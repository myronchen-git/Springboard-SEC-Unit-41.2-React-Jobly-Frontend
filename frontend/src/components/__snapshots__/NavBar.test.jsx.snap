// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavBar > matches snapshot. 1`] = `
<DocumentFragment>
  <nav
    class="NavBar"
  >
    <a
      aria-current="page"
      class="active"
      data-discover="true"
      href="/"
    >
      Home
    </a>
    <a
      class=""
      data-discover="true"
      href="/companies"
    >
      Companies
    </a>
    <a
      class=""
      data-discover="true"
      href="/jobs"
    >
      Jobs
    </a>
    <a
      class=""
      data-discover="true"
      href="/profile"
    >
      Profile
    </a>
    <a
      aria-current="page"
      class="active"
      data-discover="true"
      href="/"
    >
      Logout testuser
    </a>
  </nav>
</DocumentFragment>
`;
